### PR TITLE
feat(npm)!: drop transitiveRemediation option

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Remove transitiveRemediation option, as it only works for deprecated npm versions.

BREAKING CHANGE: transitiveRemediation option is removed, will now be ignored.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-experimental.md [hunk 1]
```diff
@@ -93,10 +93,6 @@ If set to any value, Renovate will use the Docker Hub API (`https://hub.docker.c
 If set to an integer, Renovate will use this as max page number for docker tags lookup on docker registries, instead of the default 20 pages.
 This is useful for registries which ignores the `n` parameter in the query string and only return 50 tags per page.
 
-## `RENOVATE_X_EAGER_GLOBAL_EXTENDS`
```
Comment (secustor) [inline docs/usage/self-hosted-experimental.md:96]: Accidental removal?

File: lib/modules/manager/pep621/extract.spec.ts [hunk 2]
```diff
@@ -134,14 +134,14 @@ describe('modules/manager/pep621/extract', () => {
           datasource: 'pypi',
           depType: 'project.optional-dependencies',
           currentValue: '>12',
-          depName: 'pytest/pytest',
```
Comment (secustor) [inline lib/modules/manager/pep621/extract.spec.ts:137]: These are changes from a commit for pep621 grouping changes

Comment (rarkins) [reply lib/modules/manager/pep621/extract.spec.ts]: I think these were rebase-related and now resolved

**Review by secustor — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by RahulGautamSingh** [comment]
As a part of this PR these code blocks neeed to be removed too I think.

https://github.com/renovatebot/renovate/blob/8ce089bacb0fa9d2ee4258e47cea0178535607b3/lib/workers/repository/update/branch/get-updated.ts#L97-L128

https://github.com/renovatebot/renovate/blob/8ce089bacb0fa9d2ee4258e47cea0178535607b3/lib/workers/repository/updates/flatten.ts#L162-L193

**Comment by rarkins** [comment]
@SchroederSteffen essentially that's it. It was designed at the time for the npm v1 lock file. npm underwent some pretty big rewrites around the time of the v2 lock file, and some missing/changed capabilities in npm meant that we couldn't use some of the same tricks for updating lock files so v2 wasn't able to be immediately done. Years have passed and there hasn't been any contribution or interest in improving it from the community, either in npm or other managers, and now we're left with a feature which only works with a deprecated version of npm

**Comment by SchroederSteffen** [comment]
Thanks for the detailed response!
Still, I think it's really unfortunate that Renovate doesn't support such transitive remediations, as most vulnerabilities affect transient and not direct dependencies. (I'm currently not able to contribute it myself, though.)

**Pull request merged**